tuf/repository_lib.py

```diff
@@ -1458,9 +1489,20 @@ def generate_snapshot_metadata(metadata_directory, version, expiration_date,
       # list these roles found in the metadata directory.
       if tuf.roledb.role_exists(rolename, repository_name) and \
           rolename not in ['root', 'snapshot', 'timestamp', 'targets']:
-        fileinfodict[metadata_name] = get_metadata_versioninfo(rolename,
+
+        length, hashes = securesystemslib.util.get_file_details(
```

Comment:
Nit: we're computing length and hashes here, even if we will never use them. Further, because we could be using length, or hashes,  or both, it's suboptimal to compute both in the same function.

It's probably worth at least adding a comment here to point out the above?